test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMessageWriterTests.cs

```diff
@@ -965,6 +993,160 @@ public async Task WriteMetadataDocumentAsync_WorksForXmlCsdl()
                 "</edmx:Edmx>", payload);
         }
 
+        #region "DisposeAsync"
+#if NETCOREAPP3_1_OR_GREATER
+
+        [Fact]
+        public async Task DisposeAsync_Should_Dispose_Stream_Asynchronously()
+        {
+            // Arrange
+            AsyncStream stream = new AsyncStream(new MemoryStream());
+            InMemoryMessage message = new InMemoryMessage()
+            {
+                Stream = stream
+            };
+
+            IEdmModel edmModel = GetEdmModel();
+
+            var resource = new ODataResource()
+            {
+                TypeName = "NS.Customer",
+                Properties = new[]
+                {
+                    new ODataProperty
+                    {
+                        Name = "Id",
+                        Value = 10
+                    },
+                    new ODataProperty
+                    {
+                        Name = "Name",
+                        Value = "John"
+                    }
+                }
+            };
+
+            ODataMessageWriterSettings writerSettings = new ODataMessageWriterSettings();
+            writerSettings.EnableMessageStreamDisposal = true;
+            writerSettings.BaseUri = new Uri("http://www.example.com/");
+            writerSettings.SetServiceDocumentUri(new Uri("http://www.example.com/"));
+            message.SetHeader("Content-Type", "application/json");
+
+            var msgWriter = new ODataMessageWriter((IODataResponseMessageAsync)message, writerSettings, edmModel);

```

Comment:
Any reason you couldn't use `await using (var msgWriter ...`?

Reply:
Fixed.